src/debug_toolkit/payloads/stack_trace.py

```diff
@@ -1,19 +1,45 @@
 def entrypoint(output_path: str):
-    import threading
-    import sys
+    import time
+    import json
 
     ALL_THREADS = bool(ALL_THREADS_PLACEHOLDER)
-    stack_trace = get_traceback(ALL_THREADS)
-    for thr in threading.enumerate():
-        if sys.version_info[0] >= 3 and sys.version_info[1] >= 8:
-            #  _native_id is only from python3.8+
-            replacement_str = f"{thr.getName()} tid:{thr._native_id}"
-        else:
-            replacement_str = f"{thr.getName()} thread identity:{thr._ident}"
-        thread_id_str = '0x{:016x}'.format(thr._ident)
-        stack_trace = stack_trace.replace(thread_id_str, replacement_str)
+    AMOUNT = int(AMOUNT_PLACEHOLDER)
+    SLEEP_DURATION_S = int(SLEEP_DURATION_S_PLACEHOLDER)
+
+    thread_stack_dumps=[]
+    for _ in range(AMOUNT):
+        stack_trace = get_traceback(ALL_THREADS)
+        formated_trace = format_stack_trace(stack_trace)
+        thread_stack_dumps.append({"time": time.time(), "trace": formated_trace})
+        time.sleep(SLEEP_DURATION_S)
     with open(output_path, "w+") as output_file:
-        output_file.write(stack_trace)
+        json_formatted_str = json.dumps(thread_stack_dumps, indent=4)
+        output_file.write(json_formatted_str)
+
+
+def format_stack_trace(stack_trace: str):
+
+        import sys
+        import threading
+        try:
+            for thr in threading.enumerate():
+                thread_identity = thr._ident
+                if not thread_identity:
+                    continue
+                if sys.version_info[0] >= 3 and sys.version_info[1] >= 8:
+                    #  _native_id is only from python3.8+
+                    replacement_str = f"{thr.getName()} tid:{thr._native_id}"
+                else:
+                    replacement_str = f"{thr.getName()} thread identity:{thread_identity}"
+
+                thread_id_str = '0x{:016x}'.format(thread_identity)
+                stack_trace = stack_trace.replace(thread_id_str, replacement_str)
+        except Exception:
+            # formatting sometimes can fail based on python version
```

Comment:
When can it fail?

Reply:
If there is an issue with the thr object, It has happened
Also older python versions dont support format so it will just return the normal stack trace